servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/single/SingleConcatWithPublisherTest.java

```diff
@@ -140,6 +152,7 @@ private void invalidThenValidRequest(long invalidN) {
     @Test
     public void request0PropagatedAfterSuccess() {
         source.onSuccess(1);
+        subscriber.request(1); // get the success from the Single
```

Comment:
consider breaking into 2 tests:

- `request(0)` delivers on `onError`
- `request(0)` after `request(>0)` delivers the item and propagates the `0` to the subscription (what is done now in this test)

Reply:
The new test that I added `invalidRequestNWithInlineSourceCompletion()` is equivalent to the first test you mention, that is why I did not add it separately.